parser/src/transforms/inlining.rs

```diff
@@ -800,10 +862,13 @@ impl<'a> Inlining<'a> {
                     self.bindings.insert(binding, binding_ty);
                     Expr::SymbolAccess(current_access)
                 }
-                // TODO: Currently, calls cannot be used as iterables, because we don't have pure functions
-                // which can produce aggregates. However, when those are added, we may want to add support
-                // for that here. This branch is set up to raise an appropriate panic if we forget to do so.
-                Expr::Call(_) => unimplemented!("calls to functions as iterables"),
+                // Expr::Call(call) => {
```

Comment:
We probably should perform a transformation earlier in the compiler pipeline that rewrites comprehension expressions with calls as iterable values, into a `let` tree, where the result of each call is bound to a new variable via `let`, and at the bottom of the let tree we place the comprehension expression, with the calls rewritten to their corresponding `let`-bound variable.

Alternatively, we could disallow calls in iterable position at the syntax level, but since we don't have any specific reason why we can't support them, that might not be desirable, aside from a little less complexity.

In either case, we would then handle `Expr::Call` here just like `Expr::Binary` and `Expr::ListComprehension` below.